tests/suites/os/tests/secureboot/index.js

```diff
@@ -1,145 +1,339 @@
+								'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
+							],
 							this.link,
 						),
-						/Key was rejected by service/,
-						'Unsigned module does not load',
-					);
-				}
+						/crypt/,
+						`${args.name} is encrypted`,
+					)
+				)
+			);
+	}
 
-				const testBootloaderIntegrity = async() => {
-					return this.worker.executeCommandInHostOS(
-						['balena', 'run', '--rm', '-v', '/mnt:/mnt', 'alpine', '/bin/sh', '-c',
-							'"apk add --update --no-cache binutils && strip /mnt/boot/EFI/BOOT/bootx64.efi"'],
-						this.link,
-					).then(() => this.worker.executeCommandInHostOS('reboot', this.link)
-					).then(() => test.resolves(
-							waitForSerialOutput(/bootx64.efi: Access Denied/),
-							'Firmware will not load bootloader that fails verification',
-						)
-					).then(resetWorker);
-				};
-
-				const testBootloaderConfigIntegrity = async () => {
-					return this.worker.executeCommandInHostOS(
-						['sed', '-i', 's/lockdown=integrity//', '/mnt/efi/EFI/BOOT/grub.cfg'],
-						this.link
-					).then(() => this.worker.executeCommandInHostOS('reboot', this.link)
-					).then(() => test.resolves(
-						/* The below pattern is the expected output when the config file
-						 * fails the signature check and the fallback console is not enabled.
-						 *
-						 * This sequence of ANSI escape codes decodes as:
-						 *
-						 *	ESC[0m - reset all modes
-						 *	ESC[37m - white foreground
-						 *	ESC[40m - black background
-						 *
-						 *	https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797
-						 *
-						 * Otherwise, the output finishes the above background color escape
-						 * sequence, clears the screen, and shows 'loading Boot0003
-						 * "balenaOS"'
-						 */
-
-							waitForSerialOutput(/\x1B\[0m\x1B\[37m\x1B\[40m$/, 3), // eslint-disable-line no-control-regex
-							'Bootloader will not load configuration that fails signature verification',
-						{ todo: 'This needs reworked for GRUB 2.12' },
-						)
-					).then(resetWorker);
-				};
-
-				if (securebootSupported) {
-					if (securebootEnabled) {
-						await testFullDiskEncryption();
-						await testModuleVerification();
-						await testBootloaderIntegrity();
-						await testBootloaderConfigIntegrity();
-					} else {
-						test.comment('Secure boot is not enabled');
-					}
-				} else {
-					test.comment('Secure boot is not supported by firmware');
+	async testModuleVerification() {
+		if (this.module.name.length != 0) {
+			await this.test.resolves(
+				this.worker.executeCommandInHostOS(`modprobe ${this.module.name}`, this.link),
+				`Module ${this.module.name} with valid signature loads`,
+			);
+		} else {
+			this.test.comment("No module with valid signature specified - skipping")
+		}
+
+		if (this.module.headersVersion.length != 0) {
+			const srcDir = `${__dirname}/kernel-module-build/`
+			await fse.copy(srcDir, this.tmpDir);
+			fs.readFile(`${this.tmpDir}/docker-compose.yml`, 'utf-8', (err,data) => {
```

Comment:
A little risky here - we're inside an `async` function, and this way of reading the file is non-blocking - if we are unlucky there is a chance that the `pushContainerToDut` happens before this is actually done. 

Probably safer to just use `fse` like you have with the copy:

```js
let data = await fse.readFile(.....)
const result = data.replace(...)
```

Reply:
Totally, thanks for catching that.